zoom_utils/zoom_attendance_report.py

```diff
@@ -0,0 +1,65 @@
+from datetime import datetime
+
+from zoom_utils import utils
+from zoom_utils.api_client import ZoomAPIClient
+from zoom_utils.constants import DATE_FORMAT
+from zoom_utils.utils import get_double_encoded_uuid
+
+
+class ZoomAttendanceReport:
+    def __init__(self, start_date, end_date, meeting_id, zoom_admin_account, utc_time_diff):
+        self.zoom_admin_account = zoom_admin_account
+        self.start_date = datetime.strptime(start_date, DATE_FORMAT)
+        self.end_date = datetime.strptime(end_date, DATE_FORMAT)
+        self.meeting_id = meeting_id
+        self.utc_time_diff = utc_time_diff
+
+    def get_report(self):
+        zoom_client = ZoomAPIClient(self.zoom_admin_account)
+        report = {}
+        meetings = []
+        meeting_instances = zoom_client.get_meeting_instances(self.meeting_id).get('meetings', [])
+        for instance in meeting_instances:
+            uuid = instance['uuid']
+            if uuid.startswith('/'):
+                uuid = get_double_encoded_uuid(uuid)
+            meeting_start_date_time = instance['start_time']
+            if self.start_date <= utils.to_date_time(self.utc_time_diff, meeting_start_date_time) <= self.end_date:
+                meeting_participant_entries = zoom_client.get_participant_report(uuid)
+                self.__get_report_of_a_meeting_users(meeting_participant_entries, meeting_start_date_time, report)
+                meetings.append(meeting_start_date_time)
+        meetings.sort()
+        # Sort dictionary based on names, following method will return list of tuples(key, value)
+        report = sorted(report.items(), key=lambda x: x[1]['name'])
+        return meetings, report
```

Comment:
implement single responsibility principle and divide code into multiple functions